### PR TITLE
Fixes for compatibility with v0.7.0-alpha.0

### PR DESCRIPTION
**Pull request by dawbarton**
This PR brings compatibility with v0.7.0-alpha.0 while also maintaining compatibility with v0.6. Most of the changes are simple deprecation fixes.

A few points to note:
* The new iteration protocol incurred a few changes (minor string processing).
* To deal with changes to the AST of colons (e.g. `x[1:3]`) I've added `:(:)` to the MATLAB infix operators. As such, `mstatement` doesn't produce such nice output as before but it works without special casing. (For example `mstatement(:(x[1:3])) == "x((1) : (3))"` rather than `"x(1:3)"` as before.)
* The mat_str macro now checks if an interpolated variable assignment is a ref (e.g., `mat"$(x[1:3]) = 3"`); in this case v0.7 requires that a broadcasted assignment is used explicitly.

Tests pass (on my machine at least) with v0.6.2 and v0.7.0-alpha.0.

### CODE REVIEW

File: src/MATLAB.jl [hunk 1]
```diff
@@ -173,16 +175,16 @@ end
...
-function nfields(mx::MxArray) 
+function nfields(mx::MxArray)
     Base.depwarn("MATLAB.nfields is deprecated, use mxnfields instead.", :nfields)
     return mxfields(mx)
 end
 
 @deprecate jvariable jvalue
 
-function jvariable(mx::MxArray, ty::Type{AbstractString}) 
-    Base.depwarn("jvariable(mx::MxArray,ty::Type{AbstractString}) is
```
Comment (musm) [inline src/MATLAB.jl]: please add back spaces to the depwarn

File: src/mxarray.jl [hunk 2]
```diff
@@ -68,7 +68,7 @@ const mxCELL_CLASS     = convert(mxClassID, 1)
 const mxSTRUCT_CLASS   = convert(mxClassID, 2)
 const mxLOGICAL_CLASS  = convert(mxClassID, 3)
 const mxCHAR_CLASS     = convert(mxClassID, 4)
-const mxVOID_CLASS     = convert(mxClassID, 5)
+const mxCvoid_CLASS     = convert(mxClassID, 5)
```
Comment (musm) [inline src/mxarray.jl]: all caps please for `Cvoid`

**Comment by musm** [comment]
thanks!

**Comment by dawbarton** [comment]
I've fixed the comments from the code review - hopefully all good to go now! :smiley:

**Pull request merged**